contracts/protocols/GnosisAuction.sol

```diff
@@ -22,46 +22,45 @@ library GnosisAuction {
 
     struct AuctionDetails {
         address oTokenAddress;
+        address gnosisEasyAuction;
         address asset;
-        address underlying;
+        uint256 oTokenPremium;
         address manager;
-        uint256 premiumDiscount;
         uint256 duration;
     }
 
-    function startAuction(
-        address gnosisEasyAuction,
-        address optionsPremiumPricer,
-        AuctionDetails memory auctionDetails
-    ) internal {
-        (uint256 optionPremium, uint256 oTokenSellAmount) =
-            setupAuctionParameters(
+    function startAuction(AuctionDetails memory auctionDetails) internal {
+        uint256 oTokenSellAmount =
+            getOTokenSellAmount(
                 auctionDetails.oTokenAddress,
-                gnosisEasyAuction,
-                auctionDetails.underlying,
-                optionsPremiumPricer,
-                auctionDetails.premiumDiscount
+                auctionDetails.gnosisEasyAuction
             );
 
         if (
             IERC20(auctionDetails.oTokenAddress).allowance(
                 address(this),
-                gnosisEasyAuction
+                auctionDetails.gnosisEasyAuction
             ) > 0
         ) {
             IERC20(auctionDetails.oTokenAddress).safeApprove(
-                gnosisEasyAuction,
+                auctionDetails.gnosisEasyAuction,
                 0
             );
         }
 
         IERC20(auctionDetails.oTokenAddress).safeApprove(
-            gnosisEasyAuction,
+            auctionDetails.gnosisEasyAuction,
             IERC20(auctionDetails.oTokenAddress).balanceOf(address(this))
         );
 
+        require(
+            auctionDetails.oTokenPremium.mul(oTokenSellAmount) <=
```

Comment:
We can store the result of the multiplication in a variable so we're not doing it again (it has the SafeMath overhead)